onnxruntime/contrib_ops/cpu/quantization/qlinear_softmax.cc

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "contrib_ops/cpu/quantization/qlinear_softmax.h"
+
+#include <cstdint>
+#include <type_traits>
+#include <utility>
+
+#include "core/common/common.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/providers/common.h"
+#include "core/providers/cpu/tensor/transpose.h"
+
+#include "core/mlas/inc/mlas.h"
+#include "core/platform/threadpool.h"
+#include "gsl/gsl-lite.hpp"
+
+namespace onnxruntime {
+namespace contrib {
+
+constexpr int OPSET13 = 13;
+
+namespace {
+
+void QlinearBuildLookupTableUint32(gsl::span<uint32_t> table,
+                                   const float x_scale,
+                                   size_t reduce_len, bool is_signed) {
+  const double qscale =
+      fmin(static_cast<double>(UINT32_MAX) / static_cast<double>(reduce_len), static_cast<double>(0x7fffff));
+  for (int32_t i = 0; i < 256; i++) {
+    double scaled_exp_xi = qscale * exp(static_cast<double>(i - 255) * static_cast<double>(x_scale));
```

Comment:
>qscale [](http://example.com/codeflow?start=27&length=6)

what's the benefit to convert it to integer comparing to just use double or float

Reply:
Sounds good to keep using Float type. Will try it in next PR.